torch/_export/converter.py

```diff
@@ -575,36 +594,68 @@ def convert_prim_If(self, node: torch._C.Node):
         assert len(inputs) == 1
         predicate = self.get_fx_value(inputs[0])
 
-        # Get union of inputs to blocks
-        arguments = set()
-        for block in node.blocks():
-            block_args = set()
-
-            # TODO: block.inputs(), not sure what theyre used for
-
-            for block_node in block.nodes():
-                for block_node_in in block_node.inputs():
-                    if block_node_in.debugName() in self.name_to_node:
-                        block_args.add(block_node_in.debugName())
+        def _identify_inputs_as_arguments(entry):
+            """
+            Identify inputs from the innermost sub-block. This is needed
+            for nested sub-blocks when the input is hidden in the nested sub-block.
+            E.g., example IR of input is hidden in the nested sub-block.
+            Graph[x.1]
+            %1 = ...
+                Block[]
+                    Block[x.1]
+                        %2 = x.1 ...
+            """
+            arguments: Set[str] = set()
+            for block in entry.blocks():
+                for block_node in block.nodes():
+                    for block_node_in in block_node.inputs():
+                        if block_node_in.debugName() in self.name_to_node:
+                            arguments.add(block_node_in.debugName())
+                    arguments = arguments.union(
+                        _identify_inputs_as_arguments(block_node)
+                    )
+            return arguments
 
-            arguments.update(block_args)
+        # Find inputs.
+        arguments = _identify_inputs_as_arguments(node)
 
         # Lift parameters as inputs.
         for block in node.blocks():
             arguments = arguments.union(self.blocks_to_lifted_attrs[block])
 
-        arguments = list(arguments)
+        arguments_orig = list(arguments)
+
+        # Rename variables that only have digit e.g., 20, which
+        # will cause error when it is embedded into a codegen function
+        # (invalid argument name). We rename if the name is not valid for
+        # code generation.
+        # E.g.,
+        #     Graph[x.1]                 Graph[x.1]
+        #     %1 = ...          -->      %1 = ... // reaming_map["1"] = "n_1"
+        #         Block[%1]                  Block[%n_1] // sub-block: name_to_node["n_1"] = fx.node
+        #         %2 = %1 ...                %2 = %1 ... // name_to_node[renaming_map["1"]]
+        arguments_renamed = []
+        for i, argument in enumerate(arguments_orig):
+            if not is_valid_for_codegen(argument):
+                prefix = self.name_to_node[argument].name  # type: ignore[union-attr]
+                argument_renamed = f"{prefix}_{argument}"
```

Comment:
I'm curious why we can't just modify the `normalize_name` function to handle the integer inputs?

Reply:
Good point. I think we can. Then we will move the renaming mapping logic to [here](https://github.com/pytorch/pytorch/pull/128386/files#diff-43f47a5a52902847cfa2599feb12166269bdf8fc2ddea670f29e34ef29aec391R659)?

Reply:
Hm, it's not clear to me why we need a rename map. We already have a mapping of torchscript node name to fx node, and the name we're changing is just the fx node's name.